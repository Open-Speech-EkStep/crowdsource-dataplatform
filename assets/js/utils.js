const { HOUR_IN_SECONDS, SIXTY } = require("./constants");

const ALL_LANGUAGES = [
    {value: "Assamese",id: "as", text: "অসমীয়া", hasLocaleText: true},
    {value: "Bengali", id: "bn", text: "বাংলা", hasLocaleText: true},
    {value: "English", id: "en", text: "English", hasLocaleText: true},
    {value: "Gujarati", id: "gu", text: "ગુજરાતી", hasLocaleText: true},
    {value: "Hindi", id: "hi", text: "हिंदी", hasLocaleText: true},
    {value: "Kannada", id: "kn", text: "ಕನ್ನಡ", hasLocaleText: true},
    {value: "Malayalam", id: "ml", text: "മലയാളം", hasLocaleText: true},
    {value: "Marathi", id: "mr", text: "मराठी", hasLocaleText: true},
    {value: "Odia", id: "or", text: "ଓଡିଆ", hasLocaleText: true},
    {value: "Punjabi", id: "pa", text: "ਪੰਜਾਬੀ", hasLocaleText: true},
    {value: "Tamil", id: "ta", text: "தமிழ்", hasLocaleText: true},
    {value: "Telugu", id: "te", text: "తెలుగు", hasLocaleText: true}];

function convertPXToVH(px) {
    return px * (100 / document.documentElement.clientHeight);
}

function setPageContentHeight() {
    const $footer = $('footer');
    const $nav = $('.navbar');
    const edgeHeightInPixel = $footer.outerHeight() + $nav.outerHeight()
    const contentHeightInVH = 100 - convertPXToVH(edgeHeightInPixel)
    $('#content-wrapper').css('min-height', contentHeightInVH + 'vh');
}

function toggleFooterPosition() {
    const $footer = $('footer');
    $footer.toggleClass('fixed-bottom')
    $footer.toggleClass('bottom')
}

function fetchLocationInfo() {
    //https://api.ipify.org/?format=json
    return fetch('https://www.cloudflare.com/cdn-cgi/trace').then(res => res.text()).then(ipAddressText => {
        const dataArray = ipAddressText.split('\n');
        let ipAddress = "";
        for (let ind in dataArray) {
            if (dataArray[ind].startsWith("ip=")) {
                ipAddress = dataArray[ind].replace('ip=', '');
                break;
            }
        }
        if (ipAddress.length !== 0) {
            return fetch(`/location-info?ip=${ipAddress}`);
        } else {
            return new Promise((resolve, reject) => {
                reject("Ip Address not available")
            })
        }
    });
}

const performAPIRequest = (url) => {
    return fetch(url).then((data) => {
        if(!data.ok) {
            throw Error(data.statusText || 'HTTP error');
        } else {
            return Promise.resolve(data.json());
        }
    });
}

const getLocaleString = function() {
    performAPIRequest('/get-locale-strings')
        .then((response) => {
            localStorage.setItem('localeString', JSON.stringify(response));
        });
}

const updateLocaleLanguagesDropdown = (language) => {
    const dropDown = $('#localisation_dropdown');
    const localeLang = ALL_LANGUAGES.find(ele => ele.value === language);
    if (language.toLowerCase() === "english" || localeLang.hasLocaleText === false) {
        dropDown.html('<a id="english" class="dropdown-item" href="/changeLocale/en">English</a>');
    } else {
        dropDown.html(`<a id="english" class="dropdown-item" href="/changeLocale/en">English</a>
        <a id=${localeLang.value} class="dropdown-item" href="/changeLocale/${localeLang.id}">${localeLang.text}</a>`);
    }
}

const calculateTime = function (totalSeconds, isSeconds = true) {
    const hours = Math.floor(totalSeconds / HOUR_IN_SECONDS);
    const remainingAfterHours = totalSeconds % HOUR_IN_SECONDS;
    const minutes = Math.floor(remainingAfterHours / SIXTY);
    const seconds = Math.round(remainingAfterHours % SIXTY);
    if (isSeconds) {
        return { hours, minutes, seconds };
    } else {
        return { hours, minutes };
    }
};

const formatTime = function (hours, minutes = 0, seconds = 0) {
    let result = "";
    if (hours > 0) {
        result += `${hours} hrs `;
    }
    if (minutes > 0) {
        result += `${minutes} min `;
    }
    if (hours === 0 && minutes === 0 && seconds > 0) {
        result += `${seconds} sec `;
    }
    return result.substr(0, result.length - 1);
};

module.exports = { setPageContentHeight, toggleFooterPosition, fetchLocationInfo, updateLocaleLanguagesDropdown ,calculateTime, formatTime, getLocaleString, performAPIRequest}
