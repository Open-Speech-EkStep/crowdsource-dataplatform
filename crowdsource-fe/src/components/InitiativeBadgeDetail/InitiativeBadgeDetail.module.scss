.heading {
  max-width: 37.5rem;
}

.medal {
  flex: 1;
  margin-right: 0.5rem;

  @include media-breakpoint-up(lg) {
    flex: none;
    margin-right: 3rem;
  }

  &:last-child {
    margin-right: 0;
  }
}

.medalImg {
  width: 7.5rem;

  @include media-breakpoint-up(md) {
    width: 15.625rem;
  }
}
