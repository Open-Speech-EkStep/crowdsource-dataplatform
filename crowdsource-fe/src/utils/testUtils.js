import { render } from '@testing-library/react';

const Wrapper = ({ children }) => children;

const customRender = (ui, options) => render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { renderHook } from '@testing-library/react-hooks';

// override render method
export { customRender as render };
