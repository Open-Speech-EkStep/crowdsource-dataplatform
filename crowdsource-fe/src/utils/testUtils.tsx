/* eslint-disable jest/no-export */
import type { ReactNode, ReactElement, ComponentType } from 'react';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render, act } from '@testing-library/react';
import { axe } from 'jest-axe';
import puppeteer from 'puppeteer';
import type { Browser } from 'puppeteer';

const Wrapper: ComponentType = ({ children }: { children?: ReactNode }) => children as ReactElement;

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { renderHook } from '@testing-library/react-hooks';

// override render method
export { customRender as render };

export const verifyAxeTest = ({ container }: { container: RenderResult['container'] }) => {
  it('should not fail an axe audit', async () => {
    let axeResult;

    await act(async () => {
      axeResult = await axe(container);
    });

    expect(axeResult).toHaveNoViolations();
  });
};

export const verifyVRTest = (title: string, url: string) => {
  describe(`${title} VR test`, () => {
    let browser: Browser;

    beforeAll(async () => {
      browser = await puppeteer.launch({
        headless: true,
        devtools: false,
        ignoreHTTPSErrors: true,
        args: ['--start-fullscreen'],
      });
    });

    it('match snapshot', async () => {
      const page = await browser.newPage();
      await page.setViewport({ width: 1920, height: 1080 });

      await page.goto(url, {
        waitUntil: 'networkidle2',
      });

      await page.addStyleTag({
        content: `
      *,
      *::after,
      *::before {
          transition-delay: 0s !important;
          transition-duration: 0s !important;
          animation-delay: -0.0001s !important;
          animation-duration: 0s !important;
          animation-play-state: paused !important;
          caret-color: transparent !important;
      }
      `,
      });

      const image = await page.screenshot({
        fullPage: true,
      });

      expect(image).toMatchImageSnapshot();
    });

    afterAll(async () => {
      await browser.close();
    });
  });
};
