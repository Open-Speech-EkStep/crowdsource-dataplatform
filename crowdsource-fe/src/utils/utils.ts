import { jsPDF } from 'jspdf';
import { i18n } from 'next-i18next';

import { ErrorStatusCode } from 'constants/errorStatusCode';
import { INITIATIVES_MAPPING } from 'constants/initiativeConstants';
import { KEYBOARD_ERROR, LANGUAGE_UNICODE, OTHER_LANGUAGE_UNICODE } from 'constants/Keyboard';
import type { InitiativeType } from 'types/InitiativeType';

import apiPaths from '../constants/apiPaths';

var platform = require('platform');

export const convertIntoHrsFormat = (data: any, isSeconds = true) => {
  const hours = Math.floor(data / 3600);
  const remainingAfterHours = data % 3600;
  const minutes = Math.floor(remainingAfterHours / 60);
  const seconds = Math.round(remainingAfterHours % 60);
  if (isSeconds) {
    return { hours, minutes, seconds };
  } else {
    return { hours, minutes };
  }
};

export const roundOffValue = (average: any, precision: number) => {
  const formattedAverage = average.toFixed(precision);
  return formattedAverage;
};

export const capitalizeFirstLetter = (strValue: string) => {
  return strValue?.charAt(0).toUpperCase() + strValue?.slice(1);
};

const translateText = (text: string) => i18n?.t(text);

export const formatTime = (hours: number, minutes = 0, seconds = 0) => {
  const hrStr = translateText('hours1');
  const minStr = translateText('minutes1');
  const secStr = translateText('seconds1');
  let result = '';
  if (hours > 0) {
    result += `${hours} ${hrStr} `;
  }
  if (minutes > 0) {
    result += `${minutes} ${minStr} `;
  }
  if (hours === 0 && minutes === 0 && seconds > 0) {
    result += `${seconds} ${secStr} `;
  }

  if (hours === 0 && minutes === 0 && seconds === 0) {
    result += `0 ${secStr} `;
  }

  return result.substr(0, result.length - 1);
};

export const convertTimeFormat = (value: any) => {
  const { hours, minutes, seconds } = convertIntoHrsFormat(Number(value) * 60 * 60);
  const data = formatTime(hours, minutes, seconds);
  return data;
};

const roundDuration = (value: number) => {
  return Math.round(value * 100) / 100;
};

export const getHoursValue = (value: any) => {
  return roundDuration(value);
};

export const getMinutesValue = (value: any) => {
  return roundDuration(value * 60);
};

export const getHoursText = (value: any) => {
  return `${getHoursValue(value)} ${i18n?.t('hours2')}`;
};

export const getMinutesText = (value: any) => {
  return `${getMinutesValue(value)} ${i18n?.t('minutes2')}`;
};

export const isSunoOrBoloInitiative = (value: InitiativeType) => {
  const boloOrSuno = ['asr', 'text'];
  return boloOrSuno.includes(value);
};

export const isBoloInitiative = (value: InitiativeType) => {
  const boloOrSuno = ['text'];
  return boloOrSuno.includes(value);
};

export const findInputError = (text: string, currentModule: string, language: string) => {
  if (!text.length) return;
  const newText = text.replace(/\s/g, ''); //read input value, and remove "space" by replace \s
  let error: any = KEYBOARD_ERROR.language;
  const isSunoModule = currentModule === INITIATIVES_MAPPING.suno;
  const specialSymbols = /[\u0021-\u002F\u003A-\u0040\u005B-\u0060\u007B-\u007E\u0964-\u0965]/;
  if (currentModule === INITIATIVES_MAPPING.suno && specialSymbols.test(newText) === true) {
    return KEYBOARD_ERROR.symbol;
  }
  Object.entries(isSunoModule ? LANGUAGE_UNICODE : OTHER_LANGUAGE_UNICODE).forEach(([key, value]) => {
    if (value.test(newText)) {
      //Check Unicode to see which one is true
      if (key.toLowerCase() === language.toLowerCase()) {
        error = KEYBOARD_ERROR.noError;
      }
    }
  });
  return error;
};

export const fetchLocationInfo = async () => {
  const result = await fetch('https://www.cloudflare.com/cdn-cgi/trace')
    .then(res => res.text())
    .then(async ipAddressText => {
      const dataArray = ipAddressText.split('\n');
      let ipAddress = '';
      for (let ind in dataArray) {
        if (dataArray[ind].startsWith('ip=')) {
          ipAddress = dataArray[ind].replace('ip=', '');
          break;
        }
      }
      if (ipAddress.length !== 0) {
        const data = await fetch(`${apiPaths.locationInfo}?ip=${ipAddress}`);
        return data && data.json();
      } else {
        return new Promise((resolve, reject) => {
          reject('Ip Address not available');
        });
      }
    });

  return result;
};

export const getDeviceInfo = () => {
  const os = platform.os;
  return os.family + ' ' + os.version;
};

export const getBrowserInfo = () => {
  return platform.name + ' ' + platform.version;
};

export const downloadBadge = (
  locale: string | undefined,
  initiative: string,
  source: string,
  badgeType: string,
  badgeId: string
) => {
  const pdf = new jsPDF();
  const img = new Image();
  img.src = `/images/${locale}/badges/${locale}_${initiative}_${badgeType}_${source}.png`;
  img.crossOrigin = 'Anonymous';
  img.onload = function () {
    pdf.addImage(img, 50, 10, 105, 130);
    pdf.save(`${badgeType}-badge.pdf`);
  };

  pdf.text(`Badge Id : ${badgeId}`, 36, 190);
};

export const groupBy = function (list: any, keyName: any) {
  if (!list || list.length === 0 || !list[0][keyName]) {
    return [];
  }
  return list.reduce(function (pair: any, key: any) {
    (pair[key[keyName]] = pair[key[keyName]] || []).push(key);
    return pair;
  }, {});
};

/* istanbul ignore next */
export const visualize = (visualizer: any, analyser: any) => {
  const canvasCtx = visualizer.getContext('2d');
  const bufferLength = analyser.frequencyBinCount;
  const dataArray = new Uint8Array(bufferLength);
  const WIDTH = visualizer.width;
  const HEIGHT = visualizer.height;
  // TODO do we need to limit the number of time visualize refreshes per second
  // so that it can run on Android processors without causing audio to drop?
  function draw() {
    // this is more efficient than calling with processor.onaudioprocess
    // and sending floatarray with each call...
    requestAnimationFrame(draw);
    analyser.getByteTimeDomainData(dataArray);
    // canvasCtx.fillStyle = 'rgb(255, 255, 255, 0)';
    // canvasCtx.fillRect(0, 0, WIDTH, HEIGHT);
    canvasCtx.clearRect(0, 0, WIDTH, HEIGHT);
    canvasCtx.lineWidth = 2;
    canvasCtx.strokeStyle = 'rgb(0,123,255)';
    canvasCtx.beginPath();
    const sliceWidth = (WIDTH * 1.0) / bufferLength;
    let x = 0;
    for (let i = 0; i < bufferLength; i++) {
      let v = dataArray[i] / 128.0; // uint8
      let y = (v * HEIGHT) / 2; // uint8
      if (i === 0) {
        canvasCtx.moveTo(x, y);
      } else {
        canvasCtx.lineTo(x, y);
      }
      x += sliceWidth;
    }
    canvasCtx.lineTo(visualizer.width, visualizer.height / 2);
    canvasCtx.stroke();
  }

  draw();
};

export const getErrorMsg = (error: any) => {
  if (error && error.status === ErrorStatusCode.TOO_MANY_REQUEST) {
    return 'multipleRequestApiError';
  } else {
    return 'apiFailureError';
  }
};
