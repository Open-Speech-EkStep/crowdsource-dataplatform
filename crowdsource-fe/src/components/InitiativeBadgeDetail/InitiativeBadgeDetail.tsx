import React, { useState, useEffect } from 'react';

import { useTranslation, Trans } from 'next-i18next';
import Col from 'react-bootstrap/Col';
import Form from 'react-bootstrap/Form';
import Row from 'react-bootstrap/Row';

import ImageBasePath from 'components/ImageBasePath';
import Medal from 'components/Medal';
import TriColorGradientBg from 'components/TriColorGradientBg';
import apiPaths from 'constants/apiPaths';
import { INITIATIVES_MEDIA_MAPPING } from 'constants/initiativeConstants';
import { LOCALE_LANGUAGES } from 'constants/localesConstants';
import { useFetchWithInit } from 'hooks/useFetch';
import type { Initiative } from 'types/Initiatives';
import { capitalizeFirstLetter } from 'utils/utils';

import styles from './InitiativeBadgeDetail.module.scss';

interface InitiativeBadgeDetailProps {
  action: string;
  initiative: Initiative;
  language: string;
  onSelectSourceType: (value: string) => void;
}

const InitiativeBadgeDetail = ({
  initiative,
  action,
  language,
  onSelectSourceType,
}: InitiativeBadgeDetailProps) => {
  const medals = ['bronze', 'silver', 'gold', 'platinum'];
  const languageCode = LOCALE_LANGUAGES[language];
  const { t } = useTranslation();

  const [participatedAction, setParticipatedAction] = useState<string>(action);
  const [selectedMedal, setSelectedMedal] = useState<string>(medals[0]);
  const { data, mutate: rewardMutate } = useFetchWithInit<Array<{ contributions: number; badge: string }>>(
    `${apiPaths.rewardInfo}?type=${INITIATIVES_MEDIA_MAPPING[initiative]}&source=${participatedAction}&language=${language}`,
    {
      revalidateOnMount: false,
    }
  );

  useEffect(() => {
    setParticipatedAction(action);
  }, [action]);

  useEffect(() => {
    rewardMutate();
  }, [participatedAction, initiative, language, rewardMutate]);

  const contributionCount = data && data[medals.indexOf(selectedMedal)].contributions;

  return (
    <div data-testid="InitiativeBadgeDetail">
      <Row className="py-5">
        <Col md="3" className="d-flex align-items-center fw-light display-3">
          {t('selectBadgeLevelPrompt')}
        </Col>
        <Col md="9" className="d-flex justify-content-between justify-content-lg-start mt-5 mt-md-0">
          {medals.map(medal => (
            <div key={medal} className={styles.medal}>
              <Medal
                initiative={initiative}
                selectedMedal={selectedMedal}
                medal={medal}
                action={participatedAction}
                language={language}
                handleClick={() => setSelectedMedal(medal)}
              />
            </div>
          ))}
        </Col>
      </Row>
      <Row className="py-5">
        <Col md="3" className="d-flex align-items-center fw-light display-3">
          {t('participation')}
        </Col>
        <Col md="9" className="d-flex justify-content-between justify-content-md-start mt-5 mt-md-0">
          <Form.Group className="fw-light display-3" controlId="action">
            <Form.Check
              inline
              type="radio"
              id="action1"
              data-testid="action1"
              label={t('contribution')}
              value="Contribution"
              name="action"
              className="me-9"
              checked={participatedAction == 'contribute'}
              onChange={() => {
                setParticipatedAction('contribute');
                onSelectSourceType('contribute');
              }}
            />
            <Form.Check
              inline
              type="radio"
              id="action2"
              data-testid="action2"
              label={t('validation')}
              value="Validation"
              name="action"
              checked={participatedAction == 'validate'}
              onChange={() => {
                setParticipatedAction('validate');
                onSelectSourceType('validate');
              }}
            />
          </Form.Group>
        </Col>
      </Row>
      <section className="py-5">
        <TriColorGradientBg>
          <Row>
            <Col
              xs={{ order: 1, span: 12 }}
              md={{ order: 2, span: 4 }}
              className="d-flex align-items-center justify-content-center"
            >
              <div className={styles.medalImg}>
                <ImageBasePath
                  src={`/images/${languageCode}/badges/${languageCode}_${initiative}_${selectedMedal}_${participatedAction.toLowerCase()}.svg`}
                  width="250"
                  height="320"
                  alt={`Bronze Badge ${language}`}
                />
              </div>
            </Col>
            <Col
              xs={{ order: 2, span: 12 }}
              md={{ order: 1, span: 8 }}
              className="d-flex align-items-center justify-content-center text-center text-md-start mt-5 mt-md-0"
            >
              <div className={styles.heading}>
                <h2>
                  <Trans
                    i18nKey={`badgeDetailText`}
                    defaults={`badgeDetailText`}
                    values={{
                      action: t(participatedAction),
                      initiativeName: capitalizeFirstLetter(`${t(initiative)} ${t('india')}`),
                      badge: capitalizeFirstLetter(t(selectedMedal.toLowerCase())),
                      language: capitalizeFirstLetter(t(language.toLowerCase())),
                      sourceType: t(`${INITIATIVES_MEDIA_MAPPING[initiative]}SourceType`).toLocaleLowerCase(),
                      count: contributionCount,
                    }}
                    components={{ span: <span className="text-warning" /> }}
                  />
                </h2>
              </div>
            </Col>
          </Row>
        </TriColorGradientBg>
      </section>
    </div>
  );
};

export default InitiativeBadgeDetail;
