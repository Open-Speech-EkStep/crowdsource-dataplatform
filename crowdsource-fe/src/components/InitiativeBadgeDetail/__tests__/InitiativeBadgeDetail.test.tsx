import { render, verifyAxeTest, screen, userEvent } from 'utils/testUtils';

import InitiativeBadgeDetail from '../InitiativeBadgeDetail';

describe('InitiativeBadgeDetail', () => {
  const setup = async (
    initiative: 'suno' | 'likho' | 'bolo' | 'dekho',
    language: string,
    action: 'contribute' | 'validate',
    badge: 'bronze' | 'silver' | 'gold' | 'platinum'
  ) => {
    const renderResult = render(
      <InitiativeBadgeDetail
        initiative={initiative}
        language={language}
        action={action}
        badge={badge}
        onSelectSourceType={() => {}}
        onSelectBadgeType={() => {}}
      />
    );

    return renderResult;
  };

  async () => {
    verifyAxeTest(await setup('suno', 'english', 'contribute', 'bronze'));
  };

  it('should render the component and matches it against stored snapshot', async () => {
    const { asFragment } = await setup('suno', 'english', 'contribute', 'bronze');

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Bronze')).toBeInTheDocument();
    expect(screen.getByText('Silver')).toBeInTheDocument();
    expect(screen.getByText('Gold')).toBeInTheDocument();
    expect(screen.getByText('Platinum')).toBeInTheDocument();
    expect(screen.getByText('selectBadgeLevelPrompt')).toBeInTheDocument();
  });

  it('should activate validate badges when validate radio button is selected', async () => {
    await setup('suno', 'english', 'contribute', 'bronze');
    expect(screen.getByTestId('action2')).not.toBeChecked();
    userEvent.click(screen.getByTestId('action2'));
    expect(screen.getByTestId('action2')).toBeChecked();
    expect(screen.getByTestId('action1')).not.toBeChecked();
  });

  it('should activate contribute badges when contribute radio button is selected', async () => {
    await setup('suno', 'english', 'validate', 'bronze');
    expect(screen.getByTestId('action1')).not.toBeChecked();
    userEvent.click(screen.getByTestId('action1'));
    expect(screen.getByTestId('action1')).toBeChecked();
    expect(screen.getByTestId('action2')).not.toBeChecked();
  });

  it('should activate silver badges when silver medal is selected', async () => {
    await setup('suno', 'english', 'contribute', 'bronze');
    expect(screen.getByText('Silver')).toBeInTheDocument();
    expect(screen.getByText('Silver').parentElement).not.toHaveClass('active');
    userEvent.click(screen.getByText('Silver'));
    expect(screen.getByText('Silver').parentElement).toHaveClass('active');
  });
});
