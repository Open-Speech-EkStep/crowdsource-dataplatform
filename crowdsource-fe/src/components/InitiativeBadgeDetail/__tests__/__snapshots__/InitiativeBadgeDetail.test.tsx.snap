// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitiativeBadgeDetail should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="InitiativeBadgeDetail"
  >
    <div
      class="py-5 row"
    >
      <div
        class="d-flex align-items-center fw-light display-3 col-md-3"
      >
        selectBadgeLevelPrompt
      </div>
      <div
        class="d-flex justify-content-between justify-content-lg-start mt-5 mt-md-0 col-md-9"
      >
        <div
          class="medal"
        >
          <div
            class="position-relative"
          >
            <div
              class="root active d-flex flex-column align-items-center text-center py-2 py-md-3"
              role="button"
              tabindex="0"
            >
              <div
                class="medalImg d-flex"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTYiIGhlaWdodD0iNzIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="Medal"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </div>
              </div>
              <span
                class="display-5 mt-1 fw-light text-capitalize text-break"
              >
                Bronze
              </span>
            </div>
          </div>
        </div>
        <div
          class="medal"
        >
          <div
            class="position-relative"
          >
            <div
              class="root  d-flex flex-column align-items-center text-center py-2 py-md-3"
              role="button"
              tabindex="0"
            >
              <div
                class="medalImg d-flex"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTYiIGhlaWdodD0iNzIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="Medal"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </div>
              </div>
              <span
                class="display-5 mt-1 fw-light text-capitalize text-break"
              >
                Silver
              </span>
            </div>
          </div>
        </div>
        <div
          class="medal"
        >
          <div
            class="position-relative"
          >
            <div
              class="root  d-flex flex-column align-items-center text-center py-2 py-md-3"
              role="button"
              tabindex="0"
            >
              <div
                class="medalImg d-flex"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTYiIGhlaWdodD0iNzIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="Medal"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </div>
              </div>
              <span
                class="display-5 mt-1 fw-light text-capitalize text-break"
              >
                Gold
              </span>
            </div>
          </div>
        </div>
        <div
          class="medal"
        >
          <div
            class="position-relative"
          >
            <div
              class="root  d-flex flex-column align-items-center text-center py-2 py-md-3"
              role="button"
              tabindex="0"
            >
              <div
                class="medalImg d-flex"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTYiIGhlaWdodD0iNzIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="Medal"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </div>
              </div>
              <span
                class="display-5 mt-1 fw-light text-capitalize text-break"
              >
                Platinum
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="py-5 row"
    >
      <div
        class="d-flex align-items-center fw-light display-3 col-md-3"
      >
        participation
      </div>
      <div
        class="d-flex justify-content-between justify-content-md-start mt-5 mt-md-0 col-md-9"
      >
        <div
          class="fw-light display-3"
        >
          <div
            class="me-9 form-check form-check-inline"
          >
            <input
              checked=""
              class="form-check-input"
              data-testid="action1"
              id="action1"
              name="action"
              type="radio"
              value="Contribution"
            />
            <label
              class="form-check-label"
              for="action1"
              title=""
            >
              contribution
            </label>
          </div>
          <div
            class="form-check form-check-inline"
          >
            <input
              class="form-check-input"
              data-testid="action2"
              id="action2"
              name="action"
              type="radio"
              value="Validation"
            />
            <label
              class="form-check-label"
              for="action2"
              title=""
            >
              validation
            </label>
          </div>
        </div>
      </div>
    </div>
    <section
      class="py-5"
    >
      <div
        class="root rounded-20"
      >
        <div
          class="rounded-20 bg-light p-5"
        >
          <div
            class="row"
          >
            <div
              class="d-flex align-items-center justify-content-center order-md-2 order-1 col-md-4 col-12"
            >
              <div
                class="medalImg"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUwIiBoZWlnaHQ9IjMyMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <img
                    alt="Bronze Badge english"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </div>
              </div>
            </div>
            <div
              class="d-flex align-items-center justify-content-center text-center text-md-start mt-5 mt-md-0 order-md-1 order-2 col-md-8 col-12"
            >
              <div
                class="heading"
              >
                <h2>
                  badgeDetailText
                </h2>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
